File: C/Examples/EncryptionExample/Logger.c
#include <stdio.h>
#include "message.h"
#include "Logger.h"

void loggerLog(struct message * mes)
{
	printf("Logging: [%s]\n",*(char **)getArg(0, mes));
}

void loggerSlog(struct message * mes)
{
	printf("Logging: [%s]\n",*(char **)getArg(0, mes));
}

File: DotNET/Examples/AOSD2006Demo/Logger.cs
using System;
using Composestar.RuntimeCore.FLIRT.Message;
using Composestar.RuntimeDotNET;

namespace EncryptionExample
{
	public class Logger
	{
		public Logger() { }

		[Composestar.RuntimeDotNET.Semantics("args.read")]
		public void log(ReifiedMessage rm)
		{
			string tmp = (string)rm.getArg(0);
			Console.WriteLine("Logging: ["+tmp+"]");
			rm.proceed();
		}

		public static void slog(String msg)
		{
			Console.WriteLine("Logging: ["+msg+"]");
		}
	}
}